visidata/loaders/pandas_freqtbl.py

```diff
@@ -92,60 +92,76 @@ def unselectRow(self, row):
     def updateLargest(self, grouprow):
         self.largest = max(self.largest, len(grouprow.sourcerows))
 
-    # @asyncthread
+    @asyncthread
     def reload(self):
         'Generate frequency table then reverse-sort by length.'
+        import pandas as pd
+
         # Conditions:
-        # (1) this only works if len(groupByCols) == 1 since it relies on value_counts()
         # (2) this assumes some amount of non-degeneracy (non-empty dataframe with non-null values)
+        # Note: visidata's base FrequencyTable bins numeric data in ranges (e.g. as a histogram).
+        # We currently don't provide support for this for PandasSheet.
         super().initCols(use_range=False)
         # columns now the same as the original table
         # raise ValueError(str([c.name for c in self.columns]))
 
         df = self.source.rows
-
-        if len(self.groupByCols) > 1:
-            raise NotImplementedError("Frequency Table for > 1 columns currently not supported for PandasSheet")
-        # TODO: does this make a copy?
-        # raise ValueError(str(type(self.groupByCols[0])) + " " + str(self.groupByCols[0].name))
-        this_column = df.loc[:, str(self.groupByCols[0].name)]
-        value_counts = this_column.value_counts()
+        
+        if len(self.groupByCols) == 1:
+            this_column = df.loc[:, str(self.groupByCols[0].name)]
+            value_counts = this_column.value_counts()
+        elif len(self.groupByCols) >= 1:
+            this_column = df.loc[:, str(self.groupByCols[0].name)]
+            # Compute cross-tabulation to get counts, and sort/remove zero-entries.
+            # Note that this is not space-efficient: the initial cross-tabulation will
+            # have space on the order of product of number of unique elements for each
+            # column, even though its possible the combinations present are sparse
+            # and most combinations have zero count.
+            value_counts = pd.crosstab(this_column, [df.df[c.name] for c in self.groupByCols[1:]])
```

Comment:
on second thought, it will likely make much more sense to use `pandas.pivot_table` for this. Possibly can even use to implement `PivotSheet`, albeit ambitious